### PR TITLE
Describe payment card support

### PR DESCRIPTION
**Pull request by zdrazil**
#### Changelog notes 

```
* Added section describing payment card support.
* Improved PaymentGatewayData description.
* Removed deprecated payment gateways.
* Improved CreditCardData description.
```

#### Check during review

- [ ] JSON example extended.
  - [ ] New properties are added to the correct place in the JSON.
- [ ] New properties in the table are added to the correct place.
- [ ] Correct formatting:
  - [ ] Spacing is consistent between titles, sections, tables, ...
  - [ ] Correct JSON format - indentation.
- [ ] DateTime properties should always be defined in ISO format.

### CODE REVIEW

File: distributor-widget/advanced-guide.md [hunk 1]
```diff
@@ -97,15 +97,9 @@ The deeplinks are also supported by standalone Distributor.
...
-* [Braintree](https://www.braintreepayments.com/)
-* [Adyen](https://www.adyen.com/home)
-* Mews Merchant
+* [PCI Proxy](https://www.pci-proxy.com)
 
 Using payment gateway is not mandatory, as reservations can be created without providing a credit card information.
 
-**Important:** PCI Security Standard requires you to use **SSL Certificate** on your website to be allowed to collect any payments info. This happens when using Braintree or Adyen gateways.
-
-#### Mews Merchant  <a id="mews-merchant"></a>
```
Comment (zdrazil) [inline distributor-widget/advanced-guide.md:108]: Is it correct that this information is outdated and can be removed?

File: distributor-api-v1/payment-gateway-data.md [hunk 2]
```diff
@@ -1,8 +1,7 @@
 # Payment Gateway Data
 
-To obtain`PaymentGatewayData`, you have to use client side encryption library provided by given payment gateway. You can find them here:
+`PaymentGatewayData` is an encoded representation of card number and CVV which is used to provide payment card data to Distributor API.
 
-* [Adyen](https://github.com/Adyen/CSE-JS)
-* [Stripe](https://stripe.com/docs/stripe.js)
-* [PCI Proxy](https://docs.pci-proxy.com/collect-and-store-cards/capture-iframes)
+You can use a combination of [PCI Proxy client side encryption library](https://docs.pci-proxy.com/collect-and-store-cards/capture-iframes) and your own custom implementation of Distributor client to obtain in.
```
Comment (zdrazil) [inline distributor-api-v1/payment-gateway-data.md:5]: Is it correct that we no long use the other ones and use only PCI Proxy?

File: distributor-api-v1/operations.md [hunk 3]
```diff
@@ -1055,9 +1053,9 @@ Gives a pricing information for the given configuration.
 
 |  | Property | Type | Description |
 | :--- | :--- | :--- | :--- |
-| `PaymentGatewayData` | string | required | Encoded credit card data obtained from the payment gateway specific library. More details [here](payment-gateway-data.md). |
-| `ObfuscatedCreditCardNumber` | string | required | Obfuscated credit card number, e.g.`411111******1111`. |
-| `Expiration` | string | required | Expiration of credit card in format `YYYY-MM`. Required with PCI Proxy payment gateway type. |
+| `PaymentGatewayData` | string | required | Encoded payment card data obtained from the payment gateway specific library. More details [here](payment-gateway-data.md). |
+| `Expiration` | string | required | Expiration of payment card in format `YYYY-MM`. |
```
Comment (zdrazil) [inline distributor-api-v1/operations.md:1057]: Is it correct that these are the field which API needs, the previous ones were outdated?

**Comment by zdrazil** [comment]
Hi, @koprivajakub  @janholinka  so this is the documentation for adding the payment cards. https://mews.myjetbrains.com/youtrack/issue/RND-15683 . 

Could you please check it out?

@jan.holinka among the other things could you also please check that I describe the things precisely. Such what `PaymentGatewayData` is or what is PCI Proxy etc.? I see it only from FE side.

File: distributor-widget/advanced-guide.md [hunk 4]
```diff
@@ -97,15 +97,9 @@ The deeplinks are also supported by standalone Distributor.
 
 Payment gateway is used to safely collect information about a customer’s credit card. A configuration is done once, when the hotel is set up. Distributor would use it automatically. Currently Distributor supports these gateways:
 
-* [Braintree](https://www.braintreepayments.com/)
```
Comment (zdrazil) [inline distributor-widget/advanced-guide.md:100]: Was this outdated?

File: distributor-api-v1/general-guidelines.md [hunk 5]
```diff
@@ -41,3 +41,71 @@ In case of any error, the returned JSON object describes the error and has the f
...
+This is the relation between the human readable names and the field names which Distributor API uses:
+
+| Human readable      | Distributor API       |
+| :------------------ | :-------------------- |
+| Card holder name    | HolderName            |
+| Card number         | PaymentGatewayData    |
+| Expiration date     | Expiration            |
+| CVV                 | PaymentGatewayData    |
+
+`Expiration` and `HolderName` need no special handling. You can just get them from the user in any way you want and send them to the Distributor API as is.
```
Comment (koprivajakub) [inline distributor-api-v1/general-guidelines.md:65]:
```suggestion
`Expiration` and `HolderName` are not sensitive data in terms of PCI DSS and plain text can be used. The expiration does need to follow a format that is described below.
```

File: distributor-api-v1/general-guidelines.md [hunk 6]
```diff
@@ -41,3 +41,71 @@ In case of any error, the returned JSON object describes the error and has the f
...
+
+`PaymentGatewayData`, which is an encoded string representing card number and CVV, needs more setup.
```
Comment (koprivajakub) [inline distributor-api-v1/general-guidelines.md:67]:
```suggestion
`PaymentGatewayData`, represents encoded string via payment card storage provider (LINK TO THE https://github.com/MewsSystems/gitbook-distributor-guide/blob/master/distributor-api-v1/payment-gateway-data.md). Those data are sensitive in terms of PCI DSS and therefore the implementation can not handle it in a plain text. 
```

File: distributor-api-v1/general-guidelines.md [hunk 7]
```diff
@@ -41,3 +41,71 @@ In case of any error, the returned JSON object describes the error and has the f
...
+| Card holder name    | HolderName            |
+| Card number         | PaymentGatewayData    |
+| Expiration date     | Expiration            |
+| CVV                 | PaymentGatewayData    |
+
+`Expiration` and `HolderName` need no special handling. You can just get them from the user in any way you want and send them to the Distributor API as is.
+
+`PaymentGatewayData`, which is an encoded string representing card number and CVV, needs more setup. 
+
+Because of security reasons you can't just let user enter payment card data such as CVV and card number and send them to Distributor API.
```
Comment (koprivajakub) [inline distributor-api-v1/general-guidelines.md:69]: I would remove it from here and describe it as mentioned in the previous comment

File: distributor-api-v1/general-guidelines.md [hunk 8]
```diff
@@ -41,3 +41,71 @@ In case of any error, the returned JSON object describes the error and has the f
...
+
+Instead you need to:
+* Collect and store payment card data securely with PCI Proxy.
+* Receive `transactionId` from PCI Proxy.
+* Use `transactionId` and other collected payment card data mentioned earlier to create and send [CreditCardData](operations.md#creditcarddata) to Distributor API.
+* Mews can then use [CreditCardData](operations.md#creditcarddata). For example for charging the user.
```
Comment (koprivajakub) [inline distributor-api-v1/general-guidelines.md:75]:
```suggestion
Using PCIProxy is preffered way how to obtain the card sensitive data (https://docs.pci-proxy.com/collect-and-store-cards/capture-iframes). Following the steps in the documentation should guide you through the process of obtaining the `PaymentGatewayData
```

File: distributor-api-v1/general-guidelines.md [hunk 9]
```diff
@@ -41,3 +41,71 @@ In case of any error, the returned JSON object describes the error and has the f
...
+| Human readable      | Distributor API       |
+| :------------------ | :-------------------- |
+| Card holder name    | HolderName            |
+| Card number         | PaymentGatewayData    |
+| Expiration date     | Expiration            |
+| CVV                 | PaymentGatewayData    |
+
+`Expiration` and `HolderName` are not sensitive data in terms of PCI DSS and plain text can be used. The expiration does need to follow a format that is described below.
+
+`PaymentGatewayData`, represents encoded string via payment card storage provider (LINK TO THE https://github.com/MewsSystems/gitbook-distributor-guide/blob/master/distributor-api-v1/payment-gateway-data.md). Those data are sensitive in terms of PCI DSS and therefore the implementation can not handle it in a plain text.
```
Comment (koprivajakub) [inline distributor-api-v1/general-guidelines.md:67]:
```suggestion
`PaymentGatewayData`, represents encoded string via [payment card storage provider](https://github.com/MewsSystems/gitbook-distributor-guide/blob/master/distributor-api-v1/payment-gateway-data.md). Those data are sensitive in terms of PCI DSS and therefore the implementation can not handle it in a plain text. 
```

File: distributor-widget/advanced-guide.md (hunk 4 above)
Comment (koprivajakub) [reply distributor-widget/advanced-guide.md:100]: as @mveith mentioned PCI Proxy is no longer a gateway but `payment card storage` so we might want to fix the title and description of this section as well

File: distributor-api-v1/general-guidelines.md (hunk 7 above)
Comment (zdrazil) [reply distributor-api-v1/general-guidelines.md]: will remove

File: distributor-api-v1/general-guidelines.md (hunk 6 above)
Comment (zdrazil) [reply distributor-api-v1/general-guidelines.md]: used with some edits

File: distributor-api-v1/general-guidelines.md (hunk 5 above)
Comment (zdrazil) [reply distributor-api-v1/general-guidelines.md]: used with some edits

File: distributor-api-v1/general-guidelines.md (hunk 8 above)
Comment (zdrazil) [reply distributor-api-v1/general-guidelines.md]: used the idea, tried to fit it with the rest of the text

File: distributor-api-v1/general-guidelines.md (hunk 9 above)
Comment (zdrazil) [reply distributor-api-v1/general-guidelines.md]: will fix link

**Comment by zdrazil** [comment]
> seems fine, I would maybe make it more succinct since there is a lot of repetitions. We should actually introduce the API and say "PaymentGatewayData are transactionId from PCI Proxy, read about PCI Proxy here LINK TO PCI PROXY DOCS". But I would keep the steps where we describe why it is so important to use PCIProxy.
> 
I've used your comments and ideas to make it more succinct while still mentioning most of the things.
So please if you can check it out once more.

File: distributor-api-v1/operations.md (hunk 3 above)
Comment (zdrazil) [reply distributor-api-v1/operations.md:1057]: in person discussion: yep

File: distributor-api-v1/payment-gateway-data.md (hunk 2 above)
Comment (zdrazil) [reply distributor-api-v1/payment-gateway-data.md:5]: in person discussion: yep

File: distributor-widget/advanced-guide.md (hunk 1 above)
Comment (zdrazil) [reply distributor-widget/advanced-guide.md:108]: in person discussion: yep

**Pull request merged**